pkg/reader/ccip.go

```diff
@@ -243,6 +235,21 @@ func (r *ccipChainReader) ExecutedMessageRanges(
 		query.KeyFilter{
 			Key: consts.EventNameExecutionStateChanged,
 			Expressions: []query.Expression{
+				query.Comparator(consts.EventAttributeSourceChain, primitives.ValueComparator{
+					Value:    source,
+					Operator: primitives.Eq,
+				}),
+				query.Comparator(consts.EventAttributeSequenceNumber, primitives.ValueComparator{
+					Value:    seqNumRange.Start(),
+					Operator: primitives.Gte,
+				}, primitives.ValueComparator{
+					Value:    seqNumRange.End(),
+					Operator: primitives.Lte,
+				}),
+				query.Comparator(consts.EventAttributeState, primitives.ValueComparator{
+					Value:    0,
+					Operator: primitives.Gt,
+				}),
```

Comment:
Do we need to add postgres indexes for any of these datawords? If not, this could be very slow.

If we do need to add them, I wonder if there's a way we could add them only to `chainlink-ccip`'s fork but not to `chainlink`?

Reply:
Yeah, probably will require some kind of detailed indexing. We have a task planned as performance work to go over all the DB queries we make from the ccipreader to better understand where we miss indexing/limits/paging etc
https://smartcontract-it.atlassian.net/browse/CCIP-3870

Reply:
However, in the long term we will need ChainReader tables to be optimized for the specific products, for instance by
* by having custom indexes per product
* or having custom indexers and tables per product (for instance, the ability to dynamically index events in dedicated tables to limit the scans sizes)